zulipterminal/model.py

```diff
@@ -559,6 +560,50 @@ def update_private_message(self, msg_id: int, content: str) -> bool:
         display_error_if_present(response, self.controller)
         return response["result"] == "success"
 
+    def toggle_topic_resolved_status(self, stream_id: int, topic_name: str) -> bool:
+        if self.initial_data.get("realm_allow_community_topic_editing", None):
+            self.controller.report_error(" Editing topic is disabled")
+        else:
+            request: Dict[str, Any] = {
+                "anchor": "newest",
+                "num_before": 1,
+                "num_after": 0,
+                "narrow": [
+                    {
+                        "operator": "stream",
+                        "operand": self.stream_dict[stream_id]["name"],
+                    },
+                    {"operator": "topic", "operand": topic_name},
+                ],
+            }
+            response = self.client.get_messages(request)
+            if response["messages"] != []:
+                time_since_msg_sent = time.time() - response["messages"][0]["timestamp"]
+                edit_time_limit = self.initial_data[
+                    "realm_community_topic_editing_limit_seconds"
+                ]
+                # Don't allow editing topic if time-limit exceeded.
+                if time_since_msg_sent <= edit_time_limit:
+                    topic_msg_id = response["messages"][0]["id"]
+                    if topic_name.startswith(RESOLVED_TOPIC_PREFIX):
+                        request = {
+                            "message_id": topic_msg_id,
+                            "topic": topic_name[2:],
+                            "propagate_mode": "change_all",
+                        }
+                    else:
+                        request = {
+                            "message_id": topic_msg_id,
+                            "topic": RESOLVED_TOPIC_PREFIX + topic_name,
+                            "propagate_mode": "change_all",
+                        }
+                    response = self.client.update_message(request)
```

Comment:
Can `model.update_stream_message()` be used here? unless you don't want to display `"You changed all messages' topic from #stream>topic to #stream>✔ topic ."`

Reply:
If you instead want to show `"#stream>topic has been resolved"` then do include `display_error_if_present(response)` and report_success("#stream>topic has been resolved") here.